cmd/clusterctl/main.go

```diff
@@ -17,6 +17,7 @@ limitations under the License.
 package main
 
 import (
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```

Comment:
In the interest of improving UX, would it be better to import all of the auth providers provider in the client-go auth package rather than just GCP? (This may cause issues with dependencies for the Azure provider, `dep` regularly had problems resolving this in the past, not sure if this is solved with `go mod`)
```suggestion
	_ "k8s.io/client-go/plugin/pkg/client/auth"
```

Reply:
done, seem to have had more luck with go mod

Reply:
ah actually, spoke too soon - CI seems to have the issue you described